.github/workflows/CI.yml

```diff
@@ -235,6 +221,26 @@ jobs:
         env:
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
 
+  codecov-test:
+    needs:
+      - gradle-tests
+      - console-python-tests
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Coverage Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: ./coverage-reports
+          pattern: coverage-reports-*
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v5
+        with:
+          fail_ci_if_error: true
```

Comment:
So we will see this step fail in the list of checks, but we will not be blocked from merging the changes.  I like this